File: okhttp/src/main/java/okhttp3/Streams.java
package okhttp3;

import java.io.Closeable;
import okio.BufferedSink;
import okio.BufferedSource;

public abstract class Streams implements Closeable {
    public final boolean client;
    public final BufferedSource source;
    public final BufferedSink sink;

    public Streams(boolean client, BufferedSource source, BufferedSink sink) {
      this.client = client;
      this.source = source;
      this.sink = sink;
    }
  }

File: okhttp/src/commonJvmAndroid/kotlin/okhttp3/internal/http1/Streams.kt
package okhttp3.internal.http1

import okio.BufferedSink
import okio.BufferedSource

// TODO: Consider renaming this interface to `Http1Streams` or similar
//  to avoid confusion with `Stream` in mockwebserver3?
interface Streams {
  val source: BufferedSource
  val sink: BufferedSink

  fun cancel()
}
